### ISSUE TITLE
endnotes: print pagenumber only once

### ISSUE THREAD

**Issue by floriandk** [maieul/ledmac#485]
Working the same way as `\Xendnumberonlyfirstinline` I'd suggest a similar mechanism for page-numbers (perhaps ``\pagenumberonlyfirst``) to make it possible to produce something like:

```
1:1   xxx] yyy, AB
      a] b, BC
      qwert] azer, C
  2   ccc] ddd, AE
  5   eee] fff, AD

[…]

2:1   uuu] + u, C
  4–6 sdf–ghj] Duis fermentum, metus quis
                  sed congue gravida, arcu dui quid et
                  ornare urna, ut imperdiet enim per
                  odio dignissim ipsum., A
  8   klm] om. ABC
```

I think this would be a good way to present a huge number of notes in a clear way.

It would also be good to have a hook or switch for line breaking and formatting of the page number, so that something like this would become possible:

<pre>
<b>Page 1</b>
1  xxx] yyy, AB
    a] b, BC
    qwert] azer, C
2  ccc] ddd, AE
5  eee] fff, AD

[…]

<b>Page 2</b>
2  uuu] + u, C
4–6  sdf–ghj] Duis fermentum, metus
                sed congue gravida, arcu dui
                ornare urna, ut imperdiet enim
                odio dignissim ipsum., A
8  klm] om. ABC
</pre>

I don't really know whether a parallel to `\Xnumberonlyfirstintwolines` would make sense, too… What do you think?

(This is a follow-up to #335)

**Comment by maieul** [comment]
I think I have to think ;-)

maybe next week, maybe next year

**Comment by maieul** [comment]
please send me a MWE

**Comment by floriandk** [comment]
```latex
\documentclass[a5paper]{article}

\usepackage[series={A},noeledsec,nofamiliar,noledgroup]{reledmac}

\Xendhangindent{4em}
\Xendinplaceofnumber[A]{2em}
\Xendnumberonlyfirstinline[A]

\Xendlemmaseparator[A]{\rbracket}

%prints no page numbers at all:
\renewcommand*{\printnpnum}[1]{} 



\lineation{page}

\begin{document}

\beginnumbering
\pstart
Lorem ipsum %
\edtext{xxx}{%
	\Aendnote{yyy, AB}}
 dolor %
\edtext{a}{%
	\Aendnote{b, BC}}
 sit amet, consectetuer %
\edtext{qwert}{%
	\Aendnote{azer, C}}
 adipiscing elit. Morbi commodo, %
\edtext{ccc}{%
	\Aendnote{ddd, AE}}
 ipsum sed pharetra gravida, orci magna rhoncus neque, id pulvinar odio lorem non turpis. Nullam sit amet enim. Suspendisse id velit vitae ligula volutpat condimentum. Aliquam %
\edtext{eee}{%
	\Aendnote{fff, AD}}
 erat volutpat. Sed quis velit. Nulla facilisi. Nulla libero. Vivamus pharetra posuere sapien. Nam consectetuer. Sed aliquam, nunc eget euismod ullamcorper, lectus nunc ullamcorper orci, fermentum bibendum enim nibh eget ipsum. Donec porttitor ligula eu dolor. Maecenas vitae nulla consequat libero cursus venenatis. Nam magna enim, accumsan eu, blandit sed, blandit a, eros.

Quisque facilisis erat a dui. Nam malesuada ornare dolor. Cras gravida, diam sit amet rhoncus ornare, erat elit consectetuer erat, id egestas pede nibh eget odio. Proin tincidunt, velit vel porta elementum, magna diam molestie sapien, non aliquet massa pede eu diam. Aliquam iaculis. Fusce et ipsum et nulla tristique facilisis. Donec eget sem sit amet ligula viverra gravida. Etiam vehicula urna vel turpis. Suspendisse sagittis ante a urna. Morbi a est quis orci consequat rutrum. Nullam egestas feugiat felis. Integer adipiscing semper ligula. Nunc molestie, nisl sit amet cursus convallis, sapien lectus pretium metus, vitae pretium enim wisi id lectus. Donec vestibulum. Etiam vel nibh. Nulla facilisi. Mauris pharetra. Donec augue. Fusce ultrices, neque id dignissim ultrices, tellus mauris dictum elit, vel lacinia enim metus eu nunc.

Proin at eros non eros adipiscing mollis. Donec semper turpis sed diam. Sed consequat ligula nec tortor. Integer eget sem. Ut vitae enim eu est vehicula gravida. Morbi ipsum ipsum, porta nec, tempor id, auctor vitae, purus. Pellentesque neque. Nulla luctus erat vitae libero. Integer nec %
\edtext{uuu}{%
	\Aendnote{+~u, C}}
 enim. Phasellus aliquam enim et tortor. Quisque aliquet, quam elementum condimentum feugiat, tellus odio consectetuer wisi, vel nonummy sem neque in elit. Curabitur eleifend wisi iaculis ipsum %
\edtext{sdf.  Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas gjh}{%
	\lemma{sdf--gjh}%
	\Aendnote{Duis fermentum, metus quis sed congue gravida, arcu dui quid et ornare urna, ut imperdiet enim per odio dignissim ipsum., A}}
. In non velit non ligula laoreet ultrices. Praesent ultricies facilisis nisl. Vivamus luctus elit sit amet mi. Phasellus pellentesque, erat %
\edtext{klm}{%
	\Aendnote{om. ABC}}
 eget elementum volutpat, dolor nisl porta neque, vitae sodales ipsum nibh in ligula. Maecenas mattis pulvinar diam. Curabitur sed leo.

Nulla facilisi. In vel sem. Morbi id urna in diam dignissim feugiat. Proin molestie tortor eu velit. Aliquam erat volutpat. Nullam ultrices, diam tempus vulputate egestas, eros pede varius leo, sed imperdiet lectus est ornare odio. Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Proin consectetuer velit in dui. Phasellus wisi purus, interdum vitae, rutrum accumsan, viverra in, velit. Sed enim risus, congue non, tristique in, commodo eu, metus. Aenean tortor mi, imperdiet id, gravida eu, posuere eu, felis. Mauris sollicitudin, turpis in hendrerit sodales, lectus ipsum pellentesque ligula, sit amet scelerisque urna nibh ut arcu. Aliquam in lacus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Nulla placerat aliquam wisi. Mauris viverra odio. Quisque fermentum pulvinar odio. Proin posuere est vitae ligula. Etiam euismod. Cras a eros.
\pend
\endnumbering

\newpage
\section*{Apparatus}

\setlength\parindent{0em}
\doendnotes{A}

\end{document}
```

produces my example -- completely without page numbers, which obviously doesn't make sense in a longer edition.

Instead of `\renewcommand*{\printnpnum}[1]{} ` I would like to say something like `\Xendnumberonlyfirstpagenum[A]` and e.g. `\Xendpagenumfont[A]{\bfseries}` and `\Xendahookpagenumber[A]{\par}` to get the second layout that I sketched above.

**Comment by floriandk** [comment]
No hurry btw -- this is a long-term project and most of the proofing can be done with the formatting as it is available now.
Many thanks for considering my suggestion! :)

**Comment by maieul** [comment]
For `\Xendpagenumfont[A]{\bfseries}` -> you should use `\Xendbeforepagenumber[A]{\textbf\bgroup p.~}` and `\Xendafterpagenumber[A]{\egroup)}` to wrap the page number in `\textbf\bgroup p.~`…`\egroup)` (added previous month in reledmac)

For `\Xendahookpagenumber[A]{\par}`, if I understand well, you want to add a paragraph break after the page number -> `\Xendafterpagenumber[A]{\endgraf}` (you can't use directly `\par` inside a reledmac setting command.)

I look for `\Xendnumberonlyfirstpagenum`.

**Comment by maieul** [comment]
cf 6c0054b32313725c5c78a69a482c6cda41c40e31

**Comment by maieul** [comment]
The three new commands will be:
- \Xendpagenumberonlyfirstinline
- \Xendpagenumberonlyfirstintwolines
- \Xendpagesymlinenum

**Comment by maieul** [comment]
ps : you have only one series. Do you know you can use the commands without the `[A]`>will apply to all your series, eg. A.

**Comment by maieul** [comment]
the branch issue485 provides new hooks normally adapted to your initial question. They are documented in 8601857

please:
- try the branch
- tell me if you think the handbook is clear

**Comment by floriandk** [comment]
Thanks a lot for the swift implementation! As requested, I ran some tests:

1. `\Xendpagenumberonlyfirst` and `\Xendpagenumberonlyfirstifsingle` work fine in my tests

2. when trying to reproduce layout no.1 above, it looks like the four `\Xendbox...`-commands don't work yet -- is that correct? And wouldn't we need also a box for the page numbers?

3. you suggest to use `\Xendbeforepagenumber[A]{\textbf\bgroup Page~}` and `\Xendafterpagenumber[A]{\egroup)}` for bold (or other formatting) and `\Xendafterpagenumber[A]{\endgraf}` for writing the page-number to an extra line. Independently these work fine; but together they give an error (`Missing } inserted.`) and extend the bold face also on the first line number.

4. with `\Xendpagenumberonlyfirst` the page number is written, when it occurs for the first time totally, also, when this is at the second linenumber at a page break, e.g. as 
<pre>
29–p. 2)1 lemma] var
1 lemma] var
</pre>
When I try to build my layout no 2, this doesn't work well. It should be
<pre>
29--1 lemma] var
p. 2
1 lemma] var
</pre>

5. the handbook text looks clear to me. But I don't think you can write "when a lemma override two pages" perhaps you could write "when a lemma spans over two pages".

**Comment by maieul** [comment]
1. Ok nice
2. Please provides me your MWE used for testing.
3. I did'nt play attention that \Xendafterpagenumber[A] was use both time. So the second setting overrides the first one. So you should do `\Xendafterpagenumber[A]{\egroup)\endgraf}`, and not `\Xendafterpagenumber[A]{\egroup)}` `\Xendafterpagenumber[A]{\endgraf}`. For you information : `\textbf{xxx}` can be also write as `\textbf\bgroup xxx\egroup`. It is useful when you have to split open and close bracket. It is not possible to do `\Xendbeforepagenumber[A]{\textbf{ Page~}`, because there unbalanced bracket, so we do `\Xendbeforepagenumber[A]{\textbf\bgroup Page~}`. As you did'nt close the group it said `(Missing } inserted.`
4. same thinkg, need MWE.
5. Corrected on c0b3f6d

**Comment by maieul** [comment]
hi @floriandk some news about the requested MWE?

**Comment by floriandk** [comment]
coming soon; I have been busy but haven't forgotten

**Comment by maieul** [comment]
hi @floriandk. Happy new year. May it be possible to have these thing !

**Comment by maieul** [comment]
hi @floriandk may I ask you again?

**Comment by maieul** [comment]
hi @floriandk hi, some news ?

**Comment by floriandk** [comment]
A happy new year to you, too! I'm sorry this here took me quite a while...
I'll split this up by my point numbers above.

**ad 1**

While working the other points below, I noticed that `\Xendpagenumberonlyfirst` gives correct but somehow confusing results in some cases. Please consider:

```latex
\documentclass[a5paper,12pt]{article}

\usepackage[series={A},noeledsec,nofamiliar,noledgroup]{reledmac}

\Xendpagenumberonlyfirst
%\Xendpagenumberonlyfirstifsingle
%\Xendpagenumberonlyfirstintwo

\begin{document}

\beginnumbering
\pstart
Lorem  %
\edtext{xxx}{%
    \linenum{1|1||1|2|}\Aendnote{\emph{1.1--1.2}}}
\edtext{xxx}{%
    \linenum{1|2||1|3|}\Aendnote{\emph{1.2--1.3}}}
\edtext{xxx}{%
    \linenum{1|4||1|4|}\Aendnote{\emph{1.4--1.4}}}
\edtext{xxx}{%
    \linenum{1|14||1|14|}\Aendnote{\emph{1.14--1.14}}}
\edtext{xxx}{%
    \linenum{1|15||1|16|}\Aendnote{\emph{1.15--1.16}}}
\edtext{xxx}{%
    \linenum{1|18||2|1|}\Aendnote{\emph{1.18--2.1}}}
\edtext{x}{%
    \linenum{1|20||2|2|}\Aendnote{\emph{1.20--2.2}}}
\edtext{xxxxx}{%
    \linenum{2|1||3|1|}\Aendnote{\emph{2.1--3.1}}}
\edtext{xxx}{%
    \linenum{3|2||3|3|}\Aendnote{\emph{3.2--3.3}}}
\pend
\endnumbering


\section*{Apparatus}

\setlength\parindent{0em}
\doendnotes{A}

\end{document}
```

The last line "3.2–3.3" is only printed "2-3" and I have to look at the second part of the line on top of it "p.2) 1–p.3) 1" to find out which page this "2-3" is on. This is no problem if I consult all of the apparatus line by line anyway. But if I only want to look up one particular variant, this may be quite puzzling.

Wouldn't it be more logical to have a result like

```
18–p.2) 1 xxx
20–p.2) 2 xxx
p.2) 1–p.3) 1 xxx
p.3) 2–3 xxx
```

i.e. the START page number is not printed if it is equal to the START page number of the previous lemma and the END page number is not printed if it is equal to the START page number of its own lemma.
So a new "from"-page would always stand on the left of the line and could be found quickly.

I would also need this behaviour to render the apparatus as in my layout 1 (see next post).


(Used together with `\Xendpagenumberonlyfirstifsingle` the system as it is now, makes sense though IMO.)

**Comment by floriandk** [comment]
**ad 2**

Part of the problem turns out just to be a typo in the docs: it says on p. 35 `\Xendboxstartlinenumalign` and ` \Xendboxendlinenumalign` but the commands in the code are actually (and correctly, I think) `\Xendboxstartlinenum` and `\Xendboxendlinenum`.

So the implementation (contrary to what I stated earlier) works, only the docs should be corrected.


With these correct commands I can get closer to my layout no 1:

```latex
\documentclass[a5paper,12pt]{article}

\usepackage[series={A},noeledsec,nofamiliar,noledgroup]{reledmac}


%%column-like layout for the line numbers:
\Xendboxlinenum{5em}
\Xendboxstartlinenum{2em}
\Xendboxendlinenum{7em}


\Xendbeforepagenumber{}

%%cannot be empty until we also get columns for page numbers:
\Xendafterpagenumber{.}
%%but if we get columns, only the first one can be empty; so we'd need to differenciate, e.g.:
%\Xendafterpagenumber[A]{}
%\Xendaftersecondpagenumber[A]{.}

\Xendpagenumberonlyfirst
%% see earlier post

\begin{document}

\beginnumbering
\pstart
Lorem  %
\edtext{xxx}{%
    \linenum{1|1||1|2|}\Aendnote{\emph{1.1--1.2}}}
\edtext{xxx}{%
    \linenum{1|2||1|3|}\Aendnote{\emph{1.2--1.3}}}
\edtext{xxx}{%
    \linenum{1|4||1|4|}\Aendnote{\emph{1.4--1.4}}}
\edtext{xxx}{%
    \linenum{1|14||1|14|}\Aendnote{\emph{1.14--1.14}}}
\edtext{xxx}{%
    \linenum{1|15||1|16|}\Aendnote{\emph{1.15--1.16}}}
\edtext{xxx}{%
    \linenum{1|18||2|1|}\Aendnote{\emph{1.18--2.1}}}
\edtext{x}{%
    \linenum{1|20||2|2|}\Aendnote{\emph{1.20--2.2}}}
\edtext{xxxxx}{%
    \linenum{2|1||3|1|}\Aendnote{\emph{2.1--3.1}}}
\edtext{xxx}{%
    \linenum{3|2||3|3|}\Aendnote{\emph{3.2--3.3}}}
\pend
\endnumbering


\section*{Apparatus}

\setlength\parindent{0em}
\doendnotes{A}

\end{document}
```

This looks not bad, but I'd prefer it to be
```
1    1–2     xxx 1.1–1.2
     2–3     xxx 1.2–1.3
     4       xxx 1.4–1.4
    14       xxx 1.14–1.14
    15–16    xxx 1.15–1.16
    18–2.1   xxx 1.18–2.1
    20–2.2   x 1.20–2.2
2    1–3.1   xxxxx 2.1–3.1
3    2–3     xxx 3.2–3.3
```
which would make it really easy to quickly find the references on the page.

I could possibly also somehow align the page numbers by using `\Xendinplaceofpagenumber` and `Xendinplaceofnumber` -- but wouldn't `\Xendboxpagenum` be the right way to go? 

If you are willing to do this, a `\Xendaftersecondpagenumber` would be the icing on the cake.



On the other hand I think also to introduce `\Xendboxstartpagenum` and `\Xendboxendpagenum` would not make sense though: As the second pagenumber only occurs rarely I wouldn't reserve an extra column for it.

**Comment by maieul** [comment]
It is to complex for me to follow in which part we are now. I close the issue. please open a new one with the next asked amelioration

**Comment by maieul** [comment]
@floriandk I see that I have never published these feature. So I will publish them in the next release of reledmac. For more new feature, please open a new issue.
